src/core_plugins/kibana/public/visualize/listing/visualize_landing_page_table.js

```diff
@@ -42,11 +74,13 @@ export class VisualizeLandingPageTable extends React.Component {
     const PAGE_SIZE = 3;
     this.props.fetch(newFilter).then((results) => {
       const items = results.hits;
+      const pagesCount = this.pager.getPagesCount(items.length);
+
       this.setState({
         items,
         isFetchingItems: false,
-        pager: new Pager(items.length, PAGE_SIZE, 1),
-        selectedIds: new SelectedIds()
+        currentPage: Math.min(this.state.currentPage, pagesCount - 1),
```

Comment:
This is a really good point. We should disambiguate whether this number is 0-indexed or 1-indexed in the various places it's referenced. The problem isn't formed clearly enough in my head for me to form an opinion. We can zoom about it if you want, or you can just take a stab at it and I can take a look in code review.